logical/response.go

```diff
@@ -130,3 +132,23 @@ func ListResponseWithInfo(keys []string, keyInfo map[string]interface{}) *Respon
 
 	return resp
 }
+
+// Respond404WithData takes a response and converts it to a raw response with a
+// 404 Status Code.
+func Respond404WithData(resp *Response, reqID string) (*Response, error) {
```

Comment:
Do you see value in making this `RespondWithStatusCode` and parameterizing the 404?